bib/cmd/bootc-image-builder/util.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+)
+
+func copyFile(src, dest string) error {
```

Comment:
I was starting to review this and wanted to mention that we want/need tests for this and then I stepped back and wondered if we cannot use one of the existing go modules for this? Maybe https://pkg.go.dev/github.com/otiai10/copy (I really don't mind which one fwiw). Wdyt? If we want to roll our own I think we need a bunch of basic unit tests and I'm happy to come back and review copy{Dir,File} :)

Reply:
So here's where I'm at with this:
- Roll our own: Good for keeping things small.  We know what we're exporting and can focus on just that.  For example we know we're building single regular files and don't need to cover cases like copying symlinks or preserving hard links or even needing to make decisions like that in the first place.  Though like you said, it'd be good to have unit tests, even though the use case is covered by the integration test.
- Include a separate package: Overkill?  It'd be nice if there was one that handled sparse files.  Feels a bit unnecessary to be pulling in packages that can handle all sorts of tree-copying situations for a single file in a subdirectory that is entirely predictable.
- Shell out to `cp`: Preserves sparseness.  It's a "dependency" of sorts but ... I mean it's `cp`.
- Run `cp` in the entrypoint (previous version of this PR): I'm still not convinced this is a bad idea even though it seems everyone else disagrees.  This is a containerised edge case we're solving and the entrypoint is there to do "the container stuff".  Just looking at the whole conversation and all the work we're doing to solve this is starting to feel a bit silly.